sw/device/silicon_creator/mask_rom/mask_rom_start.S

```diff
@@ -132,6 +132,16 @@ _mask_rom_start_boot:
    * need to be careful.
    */
 
+  // Check if we should halt here.
+  li   a0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
```

Comment:
curious question, what's the difference between using `li` here vs `la`?

Reply:
`li` materializes ("loads") a constant using a sequence of instructions like LUI+ADD, as needed for the particular constant. The fragments of the constant are encoded in the instruction immediate operands/fields.
`la` loads an address. The instruction sequence depends on the code model. You'd use this one instead when the operand is a symbol, something that has an address.

Reply:
Thanks for the explanation @luismarques!

https://github.com/riscv-non-isa/riscv-asm-manual/blob/master/riscv-asm.md is also helpful in explaining the [`li`](https://github.com/riscv-non-isa/riscv-asm-manual/blob/master/riscv-asm.md#load-immediate) (`lui` + `addi`) and [`la`](https://github.com/riscv-non-isa/riscv-asm-manual/blob/master/riscv-asm.md#load-address) (which is an alias for `lla` for non-PIC and `lga` for PIC) pseudo instructions.

Of course @luismarques already know about this page because he is one of the contributors :)